src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets

```diff
@@ -38,11 +38,12 @@
     DependsOnTargets="_CreateResponseFileForMvcRazorPrecompile"
     Condition="'$(TargetFrameworkIdentifier)'=='.NETCoreApp'">
 
-    <GetDotNetHost Condition="'$(MvcRazorRunCommand)'==''">
+    <GetDotNetHost Condition="'$(MSBuildRuntimeType)'=='Core' AND '$(MvcRazorRunCommand)'==''">
```

Comment:
Add a comment about all of this because there's a lot of context needed to know why this is necessary.